model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -433,6 +463,153 @@ def __init__(
         self.ntl1 = 0
         self.ntl2 = 0
 
+        self.compute_theta_and_exner = compute_theta_and_exner.with_backend(self._backend)
```

Comment:
I did not think of this in the other PRs, but can you rename all these functions from `self.foo` to `self._foo` to mark them as private? So for example for this one would become
 ```suggestion
        self._compute_theta_and_exner = compute_theta_and_exner.with_backend(self._backend)
```

And so one, for all of them.